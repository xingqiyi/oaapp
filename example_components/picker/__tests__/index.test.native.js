import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import { district } from 'antd-mobile-demo-data';
import Picker from '../index';
import List from '../../list/index';

describe('Picker', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(
      <List>
        <Picker data={district}>
          <List.Item arrow="horizontal">选择</List.Item>
        </Picker>
      </List>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
