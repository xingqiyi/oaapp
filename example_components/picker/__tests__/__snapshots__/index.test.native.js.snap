// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker renders correctly 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderTopColor": "#dddddd",
        "borderTopWidth": 0.5,
      }
    }
  >
    <View
      style={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "flexDirection": "row",
                "flexGrow": 1,
                "paddingLeft": 15,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "#dddddd",
                  "borderBottomWidth": 0.5,
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 44,
                  "paddingBottom": 6,
                  "paddingRight": 15,
                  "paddingTop": 6,
                },
                false,
                false,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 17,
                      "lineHeight": 19,
                      "textAlignVertical": "center",
                    },
                  ]
                }
              >
                选择
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#888888",
                      "fontSize": 17,
                      "lineHeight": 19,
                      "textAlign": "right",
                      "textAlignVertical": "center",
                    },
                  ]
                }
              >
                请选择
              </Text>
            </View>
            <Image
              source={1}
              style={
                Object {
                  "height": 13,
                  "marginLeft": 8,
                  "marginTop": 3,
                  "width": 8,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
        ]
      }
    />
  </View>
</View>
`;
